dev/WebView2/WebView2.h

```diff
@@ -0,0 +1,278 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#pragma once
+
+#include "pch.h"
+#include "common.h"
+
+#include <winrt\Microsoft.Web.WebView2.Core.h>
+
+namespace winrt
+{
+    using namespace Microsoft::Web::WebView2::Core;
+}
+
+#include "CoreWebView2InitializedEventArgs.g.h"
+
+#include "WebView2.g.h"
+#include "WebView2.properties.h"
+#pragma warning( push )
+#pragma warning( disable : 28251 26812)   // warning C28251: Inconsistent annotation for function: this instance has an error
+#include <webview2.h>
+#include <WebView2EnvironmentOptions.h>
+#pragma warning( pop ) 
+
+ // for making async/await possible
+struct AsyncWebViewOperations final : public Awaitable
+{
+    AsyncWebViewOperations() = default;
+
+    void StopWaiting()
+    {
+        // async operation completed, stop all awaits
+        CompleteAwaits();
+    }
+};
+
+class CoreWebView2InitializedEventArgs :
+    public winrt::implementation::CoreWebView2InitializedEventArgsT<CoreWebView2InitializedEventArgs>
+{
+public:
+    CoreWebView2InitializedEventArgs(winrt::hresult Exception) : m_exception(Exception) {};
+
+    void Exception(winrt::hresult exception) { m_exception = exception; };
+    winrt::hresult Exception() { return m_exception; }
+private:
+    winrt::hresult m_exception{};
+};
+
+template <typename D, typename T, typename ... I>
+struct __declspec(empty_bases) DeriveFromContentControlHelper_base : winrt::Windows::UI::Xaml::Controls::ContentControlT<D, winrt::default_interface<T>, winrt::composable, I...>
+{
+    using composable = T;
+    using class_type = typename T;
+
+    operator class_type() const noexcept
+    {
+        return static_cast<winrt::IInspectable>(*this).as<class_type>();
+    }
+
+    hstring GetRuntimeClassName() const
+    {
+        return hstring{ winrt::name_of<T>() };
+    }
+};
+
+class WebView2 :
+    public ReferenceTracker<WebView2, DeriveFromContentControlHelper_base, winrt::WebView2>,
```

Comment:
My C++-fu is weak; `DeriveFromContentControlHelper_base` is a template above on line 51, but here you're not specializing it on anything. How's that work?

Reply:
Incredibly, C++ lets you have a template parameter that is an unspecialized template. ReferenceTracker's template is:

```c++
template <typename D, template <typename, typename ...> class ImplT, typename ... I>
struct ReferenceTracker : public ImplT<D, I ..., ::IReferenceTrackerExtension>, public ITrackerHandleManager
```

So you can see that second template parameter matches what `winrt::implements<>` or arbitrary `winrt::DerivableTypeT` things so you can ReferenceTracker-ify any of those things.